subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheGradlePropertiesIntegrationTest.groovy

```diff
@@ -179,15 +180,53 @@ class ConfigurationCacheGradlePropertiesIntegrationTest extends AbstractConfigur
         configurationCache.assertStateStored()
 
         when:
-        System.clearProperty('fromPropertiesFile')
+        System.clearProperty("$systemProp")
         configurationCacheRun fixture.task
 
         then:
-        outputDoesNotContain "because system property 'fromPropertiesFile' has changed"
+        outputDoesNotContain "because system property '$systemProp' has changed"
         outputContains fixture.expectedOutputBeforeChange
         configurationCache.assertStateLoaded()
 
         cleanup:
-        System.clearProperty('fromPropertiesFile')
+        System.clearProperty("$systemProp")
+    }
+
+    def "passing cli override invalidates cache entry only if property was read at configuration time"() {
```

Comment:
Can we maybe split this test into two:
- one that has a read at configuration time to check if the value is expected and cache is properly invalidated.
- one that has no reads, and just checks that adding the `-D` doesn't invalidate the cache.

I think this would simplify the assertions.